src/AppInitializer.tsx

```diff
@@ -31,217 +35,245 @@ export type SignInProps = {
 
 const COOKIE_CONFIG_KEY = 'org.sagebionetworks.security.cookies.portal.config'
 
-class AppInitializer extends React.Component<Props, AppInitializerState> {
-  constructor(props: Props) {
-    super(props)
-    this.state = {
-      token: undefined,
-      hasCalledGetSession: false,
-      userProfile: undefined,
-      showLoginDialog: false,
-      synapseRedirectUrl: undefined,
+/** On mount, update the document title and meta description using data from the portal config */
+function useSetDocumentMetadataFromConfig() {
+  useEffect(() => {
+    if (document.title !== docTitleConfig.name) {
+      document.title = docTitleConfig.name
     }
-    this.updateSynapseCallbackCookie =
-      this.updateSynapseCallbackCookie.bind(this)
-  }
+    document
+      .querySelector('meta[name="description"]')!
+      .setAttribute('content', docTitleConfig.description)
+  }, [])
+}
 
-  resetSession = () => {
-    SynapseClient.signOut(this.getSession)
-    this.setState({
-      showLoginDialog: false,
-    })
-  }
+/**
+ * State and helpers for managing a user session in the portal
+ * @param setShowLoginDialog
+ * @returns
+ */
+function useSession(
```

Comment:
This could be cleaned up further, but would take some thought that I haven't given it yet